src/main/java/org/elasticsearch/common/joda/Joda.java

```diff
@@ -133,6 +134,10 @@ public static FormatDateTimeFormatter forPattern(String input, Locale locale) {
             formatter = ISODateTimeFormat.yearMonth();
         } else if ("yearMonthDay".equals(input) || "year_month_day".equals(input)) {
             formatter = ISODateTimeFormat.yearMonthDay();
+        } else if ("epochSecond".equals(input) || "epoch_second".equals(input)) {
```

Comment:
Why would we add a *new* setting that supports camelCase? The idea is to get rid of this weird dual behavior with settings. Why allow users to start using something they will just have to change later?